src/js/components/FileInput/FileInput.js

```diff
@@ -86,6 +99,36 @@ const FileInput = forwardRef(
     const removeRef = useRef();
     const RemoveIcon = theme.fileInput.icons.remove;
 
+    const [files, setFiles] = formContext.useFormInput({
+      name,
+      value: valueProp,
+      initialValue: [],
+      validate: () => {
```

Comment:
I think we can be a bit leaner and do something like: `validate: maxSize ? () => {...} : undefined`